hsclient/hydroshare.py

```diff
@@ -155,8 +181,7 @@ def _retrieve_and_parse(self, path):
     def _retrieve_checksums(self, path):
         file_str = self._hs_session.retrieve_string(path)
         data = {
-            quote(path): checksum
-            for checksum, path in (line.split("    ") for line in file_str.split("\n") if line)
+            quote(path): checksum for checksum, path in (line.split("    ") for line in file_str.split("\n") if line)
```

Comment:
It might not hurt to throw a comment here explaining what the nested comprehension is doing. Like yourself, when I can find a place to use them (nested comprehensions that is), I always do, but they can be a little tricky to quickly decipher.

Reply:
I'll add a comment.